builder/dockerignore/dockerignore.go

```diff
@@ -21,7 +21,8 @@ func ReadAll(reader io.ReadCloser) ([]string, error) {
 
 	for scanner.Scan() {
 		pattern := strings.TrimSpace(scanner.Text())
-		if pattern == "" {
+		// Blank lines or lines starting with # are ignored.
+		if pattern == "" || strings.HasPrefix(pattern, "#") {
```

Comment:
the `strings.HasPrefix(`.. should be before we trim spaces, otherwise we're not consistent with the description;

```
# this is a comment
            # this is now also considered a comment
```